infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -365,13 +349,13 @@ private void applyShadowStream(UUID streamId, long snapshot) {
      */
     public void applyShadowStreams() {
         long snapshot = rt.getAddressSpaceView().getLogTail();
-        clearTables();
         log.debug("Apply Shadow Streams, total={}", streamViewMap.size());
-        for (UUID uuid : streamViewMap.keySet()) {
-            applyShadowStream(uuid, snapshot);
+        for (UUID regularStreamId : streamViewMap.keySet()) {
+            applyShadowStream(regularStreamId, snapshot);
         }
         // Invalidate client cache after snapshot sync is completed, as shadow streams are no longer useful in the cache
         rt.getAddressSpaceView().invalidateClientCache();
+        replicatedStreamIds.clear();
```

Comment:
the clear happens in the apply phase but this set is built in the transfer phase.

Like we discussed the other day, if transfer phase started on 1 node but then resumed on another, this set will not get cleaned on the first node.  Am I right?

Reply:
correct, added a clear on the reset method, so its initialized on a new transfer.